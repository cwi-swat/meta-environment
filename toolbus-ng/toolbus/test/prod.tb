process PRODUCER(Start : int, Max : int) is
  let N : int
  in
     N := Start .
     ( if less(N, Max)
       then snd-msg(p(N)) .
            N := add(N, 1)
       fi
     ) * if equal(N, Max) then shutdown(done) fi
  endlet

process CONSUMER is
  let Z : int
  in
     ( rec-msg(q(Z?)) .
       if equal(mod(Z, 2), 0)
       then
          printf(Z)
       else
          tau
       fi
     ) * delta
  endlet

process TRAFO is
  let Z : int
  in
     ( rec-msg(p(Z?)) . snd-msg(q(Z)) ) * delta
  endlet

toolbus(
CONSUMER(),
PRODUCER(100,200),
%%PRODUCER(1000,1999),
%%PRODUCER(2000,2999),
%%TRAFO(),
%%TRAFO(),
TRAFO()
)

