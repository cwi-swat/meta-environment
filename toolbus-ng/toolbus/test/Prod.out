100
102
104
106
108
110
112
114
116
118
120
122
124
126
128
130
132
134
136
138
140
142
144
146
148
150
152
154
156
158
160
162
164
166
168
170
172
174
176
178
180
182
184
186
188
190
192
194
196
198
